sympy/polys/series/flint_powerseriesring.py

```diff
@@ -0,0 +1,352 @@
+from __future__ import annotations
+
+from contextlib import contextmanager
+from typing import Any, TypeVar
+
+from sympy.polys.domains import Domain, QQ, ZZ
+from sympy.polys.series.powerseriesring import _series_from_list, PowerSeriesRing
+
+try:
+    from flint import ctx, fmpq_poly, fmpq_series, fmpz_poly, fmpz_series
+except ImportError:
+    pass
+
+ZZSeries = TypeVar("ZZSeries", bound='fmpz_series | fmpz_poly')
+QQSeries = TypeVar("QQSeries", bound='fmpq_series | fmpq_poly')
+
+
+def _get_series_precision(s: ZZSeries | QQSeries) -> int:
+    """Helper function to get the precision of a series. By using the
+    representation of the ring"""
+
+    # XXX: This approach is inefficient, but as of python-flint 0.7.1, there is
+    # no alternative method to extract the precision from a series element.
+    rep = s.repr()
+    prec = int(rep.split('prec=')[1].split(')')[0].strip())
+    return prec
+
+@contextmanager
+def _global_cap(cap: int):
+    """Temporarily set the global series cap within a context."""
+    if ctx is None:
+        raise RuntimeError("Flint is not available")
+    old_cap, ctx.cap = ctx.cap, cap
+    try:
+        yield
+    finally:
+        ctx.cap = old_cap
+
+class FlintPowerSeriesRingZZ(PowerSeriesRing[ZZSeries]):
```

Comment:
This class is not generic so it should use a normal type as the type parameter to the generic protocol.